src/bootstrap/src/core/builder.rs

```diff
@@ -2221,13 +2221,8 @@ impl<'a> Builder<'a> {
         out
     }
 
-    /// Return paths of all submodules managed by git.
-    /// If the current checkout is not managed by git, returns an empty slice.
+    /// Return paths of all submodules.
     pub fn get_all_submodules(&self) -> &[String] {
-        if !self.rust_info().is_managed_git_subrepository() {
-            return &[];
-        }
-
```

Comment:
cc @Kobzol

Reply:
Looks like I messed up, I probably thought that for non managed checkouts submodules should be ignored or something like that. Not sure why I included this anymore.